### PR TITLE
feat: support for stream transport for debug purposes

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

Add new `stream` transport for the email notification processor for local debugging

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/notifications-backend-module-email/src/processor/NotificationsEmailProcessor.ts [hunk 1]
```diff
@@ -229,6 +232,7 @@ export class NotificationsEmailProcessor implements NotificationProcessor {
 
   private async sendMail(options: Mail.Options) {
     try {
+      this.logger.debug(`Sending notification email to ${options.to}`);
```
Comment (freben) [inline plugins/notifications-backend-module-email/src/processor/NotificationsEmailProcessor.ts:235]: remove?

**Review by freben — APPROVED** [review]

**Comment by drodil** [comment]
Not stale

**Comment by freben** [comment]
Just mentioning I'm good with this still

**Comment by drodil** [comment]
From my point of view the debug should stay :) Feel free to merge if you agree.

**Pull request merged**